File: EmpoweredPixels/Models/Matches/Match.cs
using System;
using System.Collections.Generic;
using EmpoweredPixels.DataTransferObjects.Matches;
using EmpoweredPixels.Models.Identity;

namespace EmpoweredPixels.Models.Matches
{
  public class Match
  {
    public Guid Id { get; set; }

    public DateTimeOffset? Created { get; set; }

    public DateTimeOffset? Started { get; set; }

    public long? CreatorUserId { get; set; }

    public MatchOptionsDto Options { get; set; }

    public virtual User User { get; set; }

    public virtual ICollection<MatchRegistration> Registrations { get; set; } = new List<MatchRegistration>();
  }
}

File: EmpoweredPixels/ClientApp/src/app/match/+models/match.ts
import { MatchOptions } from './match-options';
import { MatchRegistration } from './match-registration';
export class Match
{
    public id: string;

    public created: Date;

    public started: Date;

    public ended: boolean;

    public registrations: MatchRegistration[] = new Array();

    public options: MatchOptions;
}
